PatchForm.cs

```diff
@@ -57,13 +57,26 @@ public partial class PatchForm : Form {
 			}
 
 			string ebootModPath = Path.GetFullPath( "ebootmod/ebootMOD.exe" );
-			if ( !File.Exists( ebootModPath ) ) {
-				MessageBox.Show( this,
-					"ebootMOD could not be found at " + ebootModPath + "." + Environment.NewLine +
-					"ebootMOD is required to patch EBOOT.BIN. Please read the readme, find a copy of ebootMOD, and place it at the appropriate location." + Environment.NewLine +
-					"The patcher will still run, but will not be able to patch EBOOT.BIN until you do so.",
+			string makeSelfPath = Path.GetFullPath( "failOverflowTools/makeself" );
+			string unSelfPath = Path.GetFullPath ("failOverflowTools/unself");
+			string warningMessage = "{0} could not be found at {1}" + "." + Environment.NewLine +
+				"{0} is required to patch EBOOT.BIN. Please read the readme, find a copy of {0}, and place it at the appropriate location." + Environment.NewLine +
+			                        "The patcher will still run, but will not be able to patch EBOOT.BIN until you do so.";
+			if (!File.Exists (ebootModPath) && !File.Exists (makeSelfPath) && !File.Exists (unSelfPath)) {
+				MessageBox.Show (this,
+					String.Format(warningMessage, "ebootMOD", ebootModPath),
 					"ebootMOD not found!", MessageBoxButtons.OK, MessageBoxIcon.Warning
 				);
+			} else if (!File.Exists (makeSelfPath)) {
```

Comment:
You probably just want two cases here: One for Windows that checks for ebootmod, and one for non-Windows that checks for makeself and unself. Wouldn't want to extract makeself, restart the tool, and get a message that it *also* needs unself if it could just say that immediately, y'know?

Reply:
By the way, what about using the makeself and unself executables on Windows too? We could set an environment variable to a specific folder to solve the confusion about the key folder. It would almost eradicate all platform specific code. See https://github.com/tenshou/ps3fail0verflow/blob/master/tools.c#L390 regarding the path of the keys.